test/integration/connect/envoy/helpers.bash

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+# retry based on
+# https://github.com/fernandoacorreia/azure-docker-registry/blob/master/tools/scripts/create-registry-server
+# under MIT license.
+function retry {
+  local n=1
+  local max=$1
+  shift
+  local delay=$1
+  shift
+  while true; do
+    "$@" && break || {
+      exit=$?
+      if [[ $n -lt $max ]]; then
+        ((n++))
+        echo "Command failed. Attempt $n/$max:"
+        sleep $delay;
+      else
+        echo "The command has failed after $n attempts." >&2
+        return $exit
+      fi
+    }
+  done
+}
+
+function retry_default {
+  retry 5 1 $@
+}
+
+function echored {
+  tput setaf 1
```

Comment:
If you want these to run in circle it would probably be good to gate the tput commands if an env var isn't set to disable colorization.

Otherwise I have seen CI logs with tons of tput error output about $TERM not being defined etc.

Also you could potentially load build-support/scripts/functions.sh and then use the colorized output functions in 10-util.sh which already handle looking for a COLORIZED env var.

Reply:
I already fixed that in circle with `TERM` var. It works well enough although some junk chars I've not felt are important enough to track down. I think it's worth having color there still even with a small bit of jank. I didn't know about the COLORIZED funcs in build stuff that might make sense if these ever get more advanced than simple calls to tput.

Reply:
![image](https://user-images.githubusercontent.com/120915/56908171-b45e4f00-6a9d-11e9-9c12-3023b743c146.png)

Happy to make tweaks like that later.

Reply:
(trying `ansi` rather than `xterm` for circle as it seems that is an xterm-specifc escape sequence that is showing up there.)

Reply:
Yep `ansi` works better:

![image](https://user-images.githubusercontent.com/120915/56909221-bde8b680-6a9f-11e9-99b2-4fb7f095671d.png)